server/src-lib/Hasura/RQL/Types/Permission.hs

```diff
@@ -64,8 +70,8 @@ isUserVar :: T.Text -> Bool
 isUserVar = T.isPrefixOf "x-hasura-" . T.toLower
 
 roleFromVars :: UserVars -> Maybe RoleName
-roleFromVars =
-  fmap RoleName . getVarVal userRoleHeader
+roleFromVars uv =
```

Comment:
add a comment to describe what happens if `x-hasura-role` header is empty string